src/passes/MergeSimilarFunctions.cpp

```diff
@@ -628,8 +635,8 @@ EquivalentClass::replaceWithThunk(Builder& builder,
     callOperands.push_back(value);
   }
 
-  // TODO: make a return_call when possible?
-  auto ret = builder.makeCall(shared->name, callOperands, target->getResults());
+  auto ret = builder.makeCall(
```

Comment:
Can we not do `module->features.hasTailCall()` here, rather than at the caller? That would avoid adding the parameter.

Reply:
There's no `module` in scope here AFAIK. Can't get it from `Builder` or `EquivalentClass`.

Reply:
Oh, I see, thanks, `module` in the caller was passed as a parameter.

I guess we could pass `module` here too, but the current code seems just as good, lgtm.